pkg/systemd/overview-cards/usageCard.jsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import {
+    Card, CardHeader, CardBody, CardFooter, Button,
+    Progress, ProgressMeasureLocation, ProgressVariant,
+} from '@patternfly/react-core';
+
+import * as machine_info from "machine-info.js";
+import cockpit from "cockpit";
+
+const _ = cockpit.gettext;
+
+const UPDATE_DELAY = 5000;
+
+export class UsageCard extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = { pollingEnabled: true };
+        this.updateMemoryInfo = this.updateMemoryInfo.bind(this);
+    }
+
+    componentDidMount() {
+        this.updateMemoryInfo();
+
+        cockpit.addEventListener("visibilitychange", () => {
+            this.setState((prevState, _) => ({ pollingEnabled: !prevState.pollingEnabled }));
+        }, () => this.updateMemoryInfo());
+    }
+
+    componentWillUnmount() {
+        this.setState({ pollingEnabled: false });
+    }
+
+    updateMemoryInfo() {
+        const self = this;
+
+        if (!this.state.pollingEnabled)
+            return;
+
+        machine_info.cpu_ram_info().done(function(info) {
```

Comment:
This is new code, not copied, right? Please let's avoid the `self = this` dance and use an arrow function here instead.

Reply:
Done.